pmm-app/src/pmm-qan/panel/components/Details/Metrics/Metrics.utils.ts

```diff
@@ -1,33 +1,30 @@
-import {
-  applyFieldOverrides, DataFrame, FieldType, GrafanaTheme2, toDataFrame,
-} from '@grafana/data';
-import { HistogramResponse } from './Metrics.types';
-
-export const histogramToDataFrame = ({
-  histogram_items = [],
-}: HistogramResponse,
-theme: GrafanaTheme2): DataFrame[] => {
+import { ChartData } from 'chart.js';
+import { GrafanaTheme2 } from '@grafana/data';
+import { HistogramAPI } from './Metrics.types';
+
+export const getChartDataFromHistogramItems = (histogram_items: HistogramAPI[], theme: GrafanaTheme2): ChartData<'bar'>| undefined => {
+  const total = histogram_items ? histogram_items.reduce(
+    (previousValue, currentValue) => (
+      currentValue.frequency ? previousValue + currentValue.frequency : previousValue
+    ), 0,
+  ) : null;
+
   const ranges = histogram_items.map(({ range }) => formatRange(range));
   const frequencies = histogram_items.map(({ frequency }) => frequency || 0);
-  const fields = {
-    fields: [
-      { name: 'x', type: FieldType.string, values: ranges },
-      { name: '', type: FieldType.number, values: frequencies },
-    ],
+  const dataInPersent = total !== null
+    ? histogram_items.map(({ frequency }) => Math.round((frequency || 0 / total) * 10000) / 100)
```

Comment:
Why not just frequency * 100 / total ?
Not understanding where the 10000 comes from sorry

Reply:
10000 is a `magic number`, not the best practice w/o a comment I guess.

Reply:
10000 is a magic number, not the best practice w/o a comment I guess.

Reply:
This is one of the ways of rounding numbers in js, but yes, I think you're right, it would be nicer to use toFixed(2) or toPrecision, I will fix it